actionpack/lib/action_dispatch/journey/formatter.rb

```diff
@@ -33,6 +33,7 @@ def generate(name, options, path_parameters, parameterize = nil)
           defaults       = route.defaults
           required_parts = route.required_parts
           parameterized_parts.delete_if do |key, value|
+            next if defaults[key].nil?
```

Comment:
Combining control flow (`next` as early-return) and a boolean expression feels a little awkward. How about folding the not-nil check into the expression?